File: src/main/java/com/listaai/application/port/input/ListService.java
package com.listaai.application.port.input;

import com.listaai.application.port.input.command.CreateListCommand;
import com.listaai.domain.model.ShoppingList;
import java.util.List;

public interface ListService {
    List<ShoppingList> getAllLists(Long userId);
    ShoppingList createList(CreateListCommand command);
    void deleteList(Long listId, Long userId);
}

File: src/main/kotlin/com/listaai/application/port/input/ListService.kt
package com.listaai.application.port.input

import com.listaai.application.port.input.command.CreateListCommand
import com.listaai.domain.model.ShoppingList

interface ListService {
    fun getAll() : List<ShoppingList>
    fun save(createCommand: CreateListCommand) : ShoppingList
    fun delete(id : Long)
}